scripts/textmesh/text-mesh.js

```diff
@@ -0,0 +1,395 @@
+        }
+    }
+
+    cubicTo(px, py, p1x, p1y, p2x, p2y, maxSteps = 10) {
+        p.set(px, py);
+        p1.set(p1x, p1y);
+        p2.set(p2x, p2y);
+
+        const p0 = this.points[this.points.length - 1];
+        const dist = p0.distance(p1) + p1.distance(p2) + p2.distance(p);
+        const steps = Math.max(2, Math.min(maxSteps, dist / BEZIER_STEP_SIZE));
+        for (let i = 1; i <= steps; i++) {
+            const t = i / steps;
+
+            vTemp1.lerp(p0, p1, t);
+            vTemp2.lerp(p1, p2, t);
+            vTemp3.lerp(p2, p, t);
+
+            vTemp4.lerp(vTemp1, vTemp2, t);
+            vTemp5.lerp(vTemp2, vTemp3, t);
+
+            vTemp6.lerp(vTemp4, vTemp5, t);
+
+            this.points.push(vTemp6.clone());
+        }
+    }
+
+    inside(p) {
+        let count = 0, cur = this.points[this.points.length - 1];
+        this.points.forEach(next => {
+            const p0 = (cur.y < next.y ? cur : next);
+            const p1 = (cur.y < next.y ? next : cur);
+            if (p0.y < p.y + EPSILON && p1.y > p.y + EPSILON) {
+                if ((p1.x - p0.x) * (p.y - p0.y) > (p.x - p0.x) * (p1.y - p0.y)) {
+                    count += 1;
+                }
+            }
+            cur = next;
+        });
+        return (count % 2) !== 0;
+    }
+}
+
+
+var TextMesh = pc.createScript('textMesh');
+
+TextMesh.attributes.add('font', {
+    type: 'asset',
+    assetType: 'binary',
+    title: 'Font',
+    description: 'TTF file used as the basis for this 3D text'
+});
+TextMesh.attributes.add('text', {
+    type: 'string',
+    title: 'Text',
+    description: 'The text string to render as a 3D mesh'
+});
+TextMesh.attributes.add('alignment', {
+    type: 'number',
+    enum: [
+        { 'Left': 0 },
+        { 'Center': 1 },
+        { 'Right': 2 }
+    ],
+    title: 'Alignment',
+    description: 'Controls whether the text is centered or left or right justified'
+});
+TextMesh.attributes.add('characterSize', {
+    type: 'number',
+    default: 1,
+    title: 'Character Size',
+    description: 'The world space (maximum) height of each character'
+});
+TextMesh.attributes.add('characterSpacing', {
+    type: 'number',
+    min: 0,
+    default: 0,
+    title: 'Character Spacing',
+    description: 'Additional spacing between each character'
+});
+TextMesh.attributes.add('kerning', {
+    type: 'number',
+    min: 0,
+    max: 1,
+    default: 1,
+    title: 'Kerning',
+    description: 'Scales character pair kerning value so 0 is no kerning and 1 is full kerning'
+});
+TextMesh.attributes.add('depth', {
+    type: 'number',
+    default: 1,
+    title: 'Depth',
+    description: 'Depth of the extrusion applied to the text'
+});
+TextMesh.attributes.add('maxCurveSteps', {
+    type: 'number',
+    default: 10,
+    title: 'Max Curve Steps',
+    description: 'Maximum number of divisions applied to bezier based path in a font outline'
+});
+TextMesh.attributes.add('renderStyle', {
+    type: 'number',
```

Comment:
This would be nice for engine-only use, otherwise it throws lots of errors:

```suggestion
    type: 'number',
    default: 0,
```